crates/rpc/src/server/api.rs

```diff
@@ -325,6 +327,38 @@ impl api_server::Api for RpcService {
             Status::invalid_argument(err.as_report_context("invalid transaction"))
         })?;
 
+        // Rebuild a new ProvenTransaction with decorators removed from output notes
+        let mut builder = ProvenTransactionBuilder::new(
+            tx.account_id(),
+            tx.account_update().initial_state_commitment(),
+            tx.account_update().final_state_commitment(),
+            tx.account_update().account_delta_commitment(),
+            tx.ref_block_num(),
+            tx.ref_block_commitment(),
+            tx.fee(),
+            tx.expiration_block_num(),
+            tx.proof().clone(),
+        )
+        .account_update_details(tx.account_update().details().clone())
+        .add_input_notes(tx.input_notes().iter().cloned());
```

Comment:
This is fine for now, but ideally, we should have a less error-prone way to strip decorators from any note in a transaction. I'm not sure what this way would be - but let's create an issue about for this.